tests/input_processing/test_create_params_template.py

```diff
@@ -85,8 +101,9 @@ def next_a(a, a_transition):
         index=["a"],
     )
 
+    # _create_stochastic_transition_params expects nested format internally
     regime = RegimeMock(
-        transitions={"mock": {"next_a": next_a}},
+        transitions={"mock": {"next_a": next_a}},  # type: ignore[dict-item]
```

Comment:
Why are the user-facing transitions for the `RegimeMock` class still nested? This should not be the case.

Reply:
The `RegimeMock` class uses nested format here because `_create_stochastic_transition_params` specifically expects nested format - it iterates over `regime.transitions.items()` expecting regime names as keys. These tests call `_create_stochastic_transition_params` directly (not through the normal processing pipeline), so they need to provide the format that function expects. The function is not called through the normal `create_params_template` path.